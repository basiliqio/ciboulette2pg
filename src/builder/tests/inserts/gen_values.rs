use super::*;

#[test]
fn simple() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    builder
        .gen_rel_values(
            vec![
                Ciboulette2SqlValue::Text(Some(Cow::Borrowed(
                    "e1ba7ab3-12f9-4a70-aced-a1637b6a3c23",
                ))),
                Ciboulette2SqlValue::Text(Some(Cow::Borrowed(
                    "4ba2994f-0282-4251-8061-2f9cb92808e6",
                ))),
            ],
            "uuid",
        )
        .unwrap();
    let res = builder.build().unwrap();

    test_sql!(res);
}

#[test]
fn empty() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let err = builder.gen_rel_values(vec![], "mysimpletable").unwrap_err();

    assert_eq!(
        matches!(err, Ciboulette2SqlError::EmptyRelValue(x) if x == "mysimpletable"),
        true
    );
}
