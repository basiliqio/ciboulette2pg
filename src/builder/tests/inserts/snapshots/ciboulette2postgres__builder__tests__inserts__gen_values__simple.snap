---
source: src/builder/tests/inserts/gen_values.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
($0 :: uuid),
($1 :: uuid);
